File: repos_estudiantes_TARQUI_primera_entrega/REQUENA_OROS_ADRIAN_MARCELO/herencia/Herencia/src/ejercicio1/Persona.java
package ejercicio1;

public class Persona {
    private String nombre;
    private String apellido;
    private String ci;

    public Persona(String nombre, String apellido, String ci) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.ci = ci;
    }

    public String mostrarDatos() {
        return "Nombre: " + nombre + 
               "\nApellido: " + apellido + 
               "\nCI: " + ci;
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/HERENCIA/Ejercicio1py_java/persona.py
class Persona:
    def __init__(self, nombre, apellido, ci):
        self.nombre = nombre
        self.apellido = apellido
        self.ci = ci

    def method(self, tipo):
        print(f"Método ejecutado con tipo: {tipo}")

    def mostrar_info(self):
        print(f"{self.nombre} {self.apellido}, CI: {self.ci}")


class Cliente(Persona):
    def __init__(self, nombre, apellido, ci, nro_compra, id_cliente):
        super().__init__(nombre, apellido, ci)
        self.nro_compra = nro_compra
        self.id_cliente = id_cliente

    def mostrar_info(self):
        print(f"Cliente: {self.nombre} {self.apellido}, CI: {self.ci}, Nº Compra: {self.nro_compra}, ID Cliente: {self.id_cliente}")


class Jefe(Persona):
    def __init__(self, nombre, apellido, ci, sucursal, tipo):
        super().__init__(nombre, apellido, ci)
        self.sucursal = sucursal
        self.tipo = tipo

    def mostrar_info(self):
        print(f"Jefe: {self.nombre} {self.apellido}, CI: {self.ci}, Sucursal: {self.sucursal}, Tipo: {self.tipo}")
